### PR TITLE
Darc verify duplicated properties

### PR DESCRIPTION
**Pull request by JohnTortugo**
Relates to: https://github.com/dotnet/arcade/issues/4984

Check that no two combination of PropertyName + [cC]ondition exists in Version.props

### CODE REVIEW

**Comment by riarenas** [comment]
This looks like it could have some simple test cases in the GitFileManagerTests

**Comment by JohnTortugo** [comment]
> This looks like it could have some simple test cases in the GitFileManagerTests

I'll add some. Thanks Ricardo.

**Comment by JohnTortugo** [comment]
@riarenas - I added the tests.

File: src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/GitFileManager.cs [hunk 1]
```diff
@@ -752,6 +753,45 @@ public static void NormalizeAttributes(string directoryPath)
...
+        /// <returns>True if there are no duplicated properties.</returns>
+        public Task<bool> VerifyNoDuplicatedProperties(XmlDocument versionProps)
+        {
+            bool hasNoDuplicatedProperties = true;
+            HashSet<string> existingProperties = new HashSet<string>();
+
+            XmlNodeList propertyGroups = versionProps.GetElementsByTagName("PropertyGroup");
+            foreach (XmlNode propertyGroup in propertyGroups)
+            {
+                foreach (var property in propertyGroup.ChildNodes)
```
Comment (riarenas) [inline src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/GitFileManager.cs:769]: [nit] reduce var usage

**Pull request merged**